hepcrawl/pipelines.py

```diff
@@ -184,21 +192,34 @@ def close_spider(self, spider):
         """Post results to BROKER API."""
         from celery.utils.log import get_task_logger
         logger = get_task_logger(__name__)
+        if 'SCRAPY_JOB' not in os.environ:
+            self._cleanup(spider)
+            return
 
         if hasattr(spider, 'tmp_dir'):
             shutil.rmtree(path=spider.tmp_dir, ignore_errors=True)
 
-        if 'SCRAPY_JOB' in os.environ and self.count > 0:
+        if hasattr(spider, 'state'):
+            got_errors = len(spider.state.get('errors', [])) > 0
```

Comment:
Why not just `got_errors = spider.state.get('errors')`?

Reply:
done